src/main/java/org/apache/commons/imaging/formats/webp/chunks/WebPChunk.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.formats.webp.chunks;
+
+import org.apache.commons.imaging.ImageReadException;
+import org.apache.commons.imaging.common.BinaryFileParser;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+
+public abstract class WebPChunk extends BinaryFileParser {
+    public static final int TYPE_VP8 = 0x20385056;
+    public static final int TYPE_VP8L = 0x4C385056;
+    public static final int TYPE_VP8X = 0x58385056;
+    public static final int TYPE_ANIM = 0x4D494E41;
+    public static final int TYPE_ANMF = 0x464D4E41;
+    public static final int TYPE_ICCP = 0x50434349;
+    public static final int TYPE_EXIF = 0x46495845;
+    public static final int TYPE_XMP = 0x20504D58;
+
+    private final int type;
+    private final int size;
+    protected final byte[] bytes;
+
+    WebPChunk(int type, int size, byte[] bytes) throws ImageReadException {
+        super(ByteOrder.LITTLE_ENDIAN);
+
+        if (size != bytes.length) {
+            throw new AssertionError("Chunk size must match bytes length");
```

Comment:
Instead of `AssertionError` I think it would be best to throw the `ImageReadException`

Reply:
@kinow I think `AssertionError` should be used here.

I think `ImageReadException` should be thrown only when there is no error in commons-imaging itself, e.g. we encountered illegal input at runtime.

However, here, illegal input data will not cause this problem. An exception will be thrown only if commons-imaging itself has logical errors, so I use `AssertionError` here.

Reply:
You must be thinking of IllegalStateException, we should not throw Errors.

Reply:
I think it'd be important to be consistent, so users are not surprised when using the API. Have a look at the other parsers, @Glavo , and see what exceptions are being used. If you believe we must change it to another exception type, then we can create a follow up issue to discuss and plan how to do that in all other parsers :+1:

Reply:
> You must be thinking of IllegalStateException, we should not throw Errors.

In my understanding, here I am asserting that throwing an exception is an unreachable path.

But I thought again, maybe it is also feasible to throw `IllegalArgumentException`, because Chunk may also be constructed by users themselves.

Reply:
Which leads to consider if we should present the smallest API surface for this new code, IOW the minimal elements that should be public or protected. Once public or protected, it's set in stone within a major release line.